packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -247,43 +248,35 @@ export class XInstaller {
   /**
    * Install more plugins to Vue defined by the user.
    *
-   * @param snippet - The snippet configuration.
    * @param bus - The events bus used in the application.
    * @returns The arguments from the plugins installation to be used in Vue's constructor.
    * @internal
    */
-  protected installExtraPlugins(
-    snippet: SnippetConfig,
-    bus: XBus
-  ): Promise<VueConstructorPartialArgument> {
+  protected installExtraPlugins(bus: XBus): Promise<VueConstructorPartialArgument> {
     const vue = this.getVue();
-    return Promise.resolve(this.options.installExtraPlugins?.({ vue, snippet, bus }));
+    return Promise.resolve(
+      this.options.installExtraPlugins?.({ vue, snippet: this.snippetConfig!, bus })
+    );
   }
 
   /**
    * In the case that the `app` parameter is present in the {@link InstallXOptions}, then a new Vue
    * application is created using that app.
    *
    * @param extraPlugins - Vue plugins initialisation data.
-   * @param snippetConfig - Configuration from the client snippet.
    * @returns The Created Vue application or undefined if not created.
    *
    * @internal
    */
-  protected createApp(
-    extraPlugins: VueConstructorPartialArgument,
-    snippetConfig: SnippetConfig
-  ): Vue | undefined {
+  protected createApp(extraPlugins: VueConstructorPartialArgument): Vue | undefined {
     if (this.options.app !== undefined) {
       const vue = this.getVue();
-      this.snippetConfig = vue.observable(snippetConfig);
+      vue.observable(this.snippetConfig);
```

Comment:
According to the [docs](https://v2.vuejs.org/v2/api/#Vue-observable), It seems that `Vue.observable` directly mutates the object passed to it in Vue 2, but it won't in Vue 3, do we want to already work with the returned object or we shouldn't worry about that now?

Reply:
@mnavarroespinosa You're right. I'd change it now as we will probably forget later on when migrating to `vue3` and it might turn into a bug 😅

Reply:
Do you mean `reactive`? Yes, I agree, In vue3 as reactivity is based in Proxy objects they don't mutate the source, however I would still use `this.snippetConfig` rather than a new variable for the same reasons you mentioned.